pkg/imagefilter/formatter.go

```diff
@@ -54,6 +57,26 @@ func (*textResultsFormatter) Output(w io.Writer, all []Result) error {
 	return nil
 }
 
+type shellResultsFormatter struct{}
+
+func (*shellResultsFormatter) Output(w io.Writer, all []Result) error {
+	var errs []error
+
+	for _, res := range all {
+		if _, err := fmt.Fprintf(w, "%s --distro %s --arch %s\n",
+			res.ImgType.Name(),
+			res.Distro.Name(),
+			res.Arch.Name()); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if len(errs) > 0 {
+		return errors.Join(errs...)
+	}
+
+	return nil
```

Comment:
I believe you can do `return errors.Join(errs...)` directly here. `errors.Join` discards `nil-values` in errors and returns `nil` when it only has `nil`s in its arguments.

Since `errs` contains a bunch of `nil`s anyhow its `len` is likely always over zero.